url2bc31e3489_ramonli_RabbitMQResearch_tgz-pJ8-com_mpos_lottery_te_thirdpartyservice_amqp_protobuf_CancellationMessagePulisherJ8/src/com/mpos/lottery/te/thirdpartyservice/amqp/TeTransactionMessage.java

```diff
@@ -0,0 +1,6710 @@
+     *
+     * <pre>
+     * encryped 
+     * </pre>
+     */
+    boolean hasSerialNo();
+    /**
+     * <code>required string serial_no = 4;</code>
+     *
+     * <pre>
+     * encryped 
+     * </pre>
+     */
+    java.lang.String getSerialNo();
+    /**
+     * <code>required string serial_no = 4;</code>
+     *
+     * <pre>
+     * encryped 
+     * </pre>
+     */
+    com.google.protobuf.ByteString
+        getSerialNoBytes();
+
+    // required .GameInstance game_instance = 5;
+    /**
+     * <code>required .GameInstance game_instance = 5;</code>
+     */
+    boolean hasGameInstance();
+    /**
+     * <code>required .GameInstance game_instance = 5;</code>
+     */
+    com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.GameInstance getGameInstance();
+    /**
+     * <code>required .GameInstance game_instance = 5;</code>
+     */
+    com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.GameInstanceOrBuilder getGameInstanceOrBuilder();
+
+    // repeated .Sale.Entry entries = 6;
+    /**
+     * <code>repeated .Sale.Entry entries = 6;</code>
+     */
+    java.util.List<com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale.Entry> 
+        getEntriesList();
+    /**
+     * <code>repeated .Sale.Entry entries = 6;</code>
+     */
+    com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale.Entry getEntries(int index);
+    /**
+     * <code>repeated .Sale.Entry entries = 6;</code>
+     */
+    int getEntriesCount();
+    /**
+     * <code>repeated .Sale.Entry entries = 6;</code>
+     */
+    java.util.List<? extends com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale.EntryOrBuilder> 
+        getEntriesOrBuilderList();
+    /**
+     * <code>repeated .Sale.Entry entries = 6;</code>
+     */
+    com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale.EntryOrBuilder getEntriesOrBuilder(
+        int index);
+  }
+  /**
+   * Protobuf type {@code Sale}
+   *
+   * <pre>
+   * The sale transaction.
+   * </pre>
+   */
+  public static final class Sale extends
+      com.google.protobuf.GeneratedMessage
+      implements SaleOrBuilder {
+    // Use Sale.newBuilder() to construct.
+    private Sale(com.google.protobuf.GeneratedMessage.Builder<?> builder) {
+      super(builder);
+      this.unknownFields = builder.getUnknownFields();
+    }
+    private Sale(boolean noInit) { this.unknownFields = com.google.protobuf.UnknownFieldSet.getDefaultInstance(); }
+
+    private static final Sale defaultInstance;
+    public static Sale getDefaultInstance() {
+      return defaultInstance;
+    }
+
+    public Sale getDefaultInstanceForType() {
+      return defaultInstance;
+    }
+
+    private final com.google.protobuf.UnknownFieldSet unknownFields;
+    @java.lang.Override
+    public final com.google.protobuf.UnknownFieldSet
+        getUnknownFields() {
+      return this.unknownFields;
+    }
+    private Sale(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.